protobuf/common/context.go

```diff
@@ -0,0 +1,5 @@
+package common
+
+func NewContext(version *NodeVersion) *Context {
```

Comment:
Can we call this something something that doesn't overload/get confused with the core library `context.Context` that is also used in the code?

`Metadata`, `MsgContext`, or similar would help a lot in readability!

Reply:
I think we can choose a better name, of course. However, I would do it after being sure this feature is valid and any other objection is solved.